### PR TITLE
Make deleting stale pushers a background update

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
seems plausible

File: changelog.d/9536.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix deleting pushers when using sharded pushers.
```
Comment (richvdh) [inline changelog.d/9536.bugfix:1]: is this the same as the newsfile for the PR that introduced the delta file? if not, please could it be?

Comment (erikjohnston) [reply changelog.d/9536.bugfix:1]: It is, sorry should have linked to it in the PR description: #9479

**Pull request merged**